### ISSUE TITLE
Cloudflare will break Postal if CDN is used for Return Path domains

### ISSUE THREAD

**Issue by anhofmann** [atech/postal#159]
I'm moving the discussion from #154 because it is another problem.

**TL;DR: if you activate the CDN for the RP domains, the internet will not know how to deliver mails to postals return path.**
I found a Cloudflare support entry that explains it: https://support.cloudflare.com/hc/en-us/articles/200168876-My-email-or-mail-stopped-working-What-should-I-do-
Since these days tons of people use Cloudflare I would suggest to put a note in the dns setting wiki that tells people need to 'grey out' the RP Domains for Postal.

**The long explanation:**
@ro78 I'm a Cloudflare user too and I don't route the postal domains through their CDN (the cloud button is gray). Cloudflare is a HTTP Proxy but not am mail proxy. When I activate the CDN I don't get a MX record for the psrp domains. So right now Cloudflare will break Postal if you route the domains through their CDN. To use Cloudflare together with Postal you need to set all Return Path and MX domains to 'dns only' on Cloudflare. 

DNS query with domain setting 'dns only' (this is how it needs to be): 
dig psrp.mydomain.com MX @hugh.ns.cloudflare.com

; <<>> DiG 9.10.3-P4-Ubuntu <<>> psrp.mydomain.com MX @hugh.ns.cloudflare.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 31601
;; flags: qr aa rd; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 2
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;psrp.mydomain.com. IN	MX

;; ANSWER SECTION:
psrp.mydomain.com. 300 IN	CNAME	rp.postal.mydomain.com.
rp.postal.mydomain.com. 300 IN	MX	10 mx.postal.mydomain.com.

;; ADDITIONAL SECTION:
mx.postal.mydomain.com. 300 IN	A	83.169.21.5

DNS query with 'DNS and HTTP Proxy' (yellow cloud icon):
dig psrp.mydomain.com MX @hugh.ns.cloudflare.com

; <<>> DiG 9.10.3-P4-Ubuntu <<>> psrp.mydomain.com MX @hugh.ns.cloudflare.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 24063
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;psrp.mydomain.com. IN	MX

;; AUTHORITY SECTION:
mydomain.com.		3600	IN	SOA	hugh.ns.cloudflare.com. dns.cloudflare.com. 2024667776 10000 2400 604800 3600

**Comment by ro78** [comment]
My PSRP record is dns-only. No CDN enabled on it.
My issue is not MX record not resolved (MX record is OK for psrp), but CNAME that doesn't answer, so Postal can't validate the returnpath value and is not using psrp to send emails.

**Comment by catphish** [comment]
When you instruct CloudFlare to "proxy" a domain name (ie orange cloud mode) they point the domain to their own servers, then proxy only HTTP requests. This makes it incompatible with almost everything except a simple HTTP service. Is this not explained in CloudFlare's own documentation?

**Comment by anhofmann** [comment]
@catphish sure. But I think this is kind of a gotcha, because Cloudflare activates the CDN for A and CNAME records by default and by documenting this in the postal wiki would help people to not fall for this.

**Comment by adamcooke** [comment]
What are we documenting here @catphish ?

**Comment by catphish** [comment]
If we have documentation that explains what DNS records to create, we could add a note indicating then when using cloudflare (or any other combined CDN/DNS service), certain domain names (RP, MX, etc) should not be set as http proxy domains. I am not proposing this, merely explaining it.

**Comment by afanjul** [comment]
Yeah, we are also using cloudflare, but I did took this into account to disable the "orange clouds" before add the postal records to avoid this kind of issues...

**Comment by anhofmann** [comment]
I will not be grumpy if this proposal doesn't make it. You can close this ticket if you want.

**Comment by ro78** [comment]
Issue where CNAME record was not returned has been fixed by CloudFlare today.
